Core/include/Acts/Propagator/RiddersPropagator.ipp

```diff
@@ -69,21 +58,50 @@ auto Acts::RiddersPropagator<propagator_t>::propagate(
   using ThisResult = Result<action_list_t_result_t<
       BoundTrackParameters, typename propagator_options_t::action_list_type>>;
 
+  auto startWithoutCov = start;
+  startWithoutCov.covariance() = std::nullopt;
+
   // Propagate the nominal parameters
-  auto nominalRet = m_propagator.propagate(start, target, options);
-  if (not nominalRet.ok()) {
-    return ThisResult::failure(nominalRet.error());
+  auto result = m_propagator.propagate(startWithoutCov, target, options);
+  if (not result.ok()) {
+    return ThisResult::failure(result.error());
   }
-
   // Extract results from the nominal propagation
-  auto nominalResult = std::move(nominalRet).value();
-  const BoundVector& nominalParameters =
-      nominalResult.endParameters->parameters();
+  auto nominalResult = result.value();
+  assert(nominalResult.endParameters);
+  const auto& nominalFinalParameters = *nominalResult.endParameters;
+
+  auto jacobian = wiggleAndCalculateJacobian(options, start, nominalResult);
+  nominalResult.transportJacobian = jacobian;
 
+  if (start.covariance()) {
+    // use nominal parameters and Ridders covariance
+    auto cov = jacobian * (*start.covariance()) * jacobian.transpose();
+    // replace the covariance of the nominal result w/ the ridders covariance
+    nominalResult.endParameters = BoundTrackParameters(
+        nominalFinalParameters.referenceSurface().getSharedPtr(),
+        nominalFinalParameters.parameters(), nominalFinalParameters.charge(),
+        std::move(cov));
+  }
+
+  return ThisResult::success(std::move(nominalResult));
+}
+
+template <typename propagator_t>
+template <typename propagator_options_t, typename parameters_t,
+          typename result_t>
+auto Acts::RiddersPropagator<propagator_t>::wiggleAndCalculateJacobian(
+    const propagator_options_t& options, const parameters_t& start,
+    result_t& nominalResult) const -> Jacobian {
+  auto nominalFinalParameters = *nominalResult.endParameters;
+  // Use the curvilinear surface of the propagated parameters as target
+  const Surface& target = nominalFinalParameters.referenceSurface();
+
+  // TODO add to propagator options
   // Steps for estimating derivatives
-  std::vector<double> deviations = {-4e-4, -2e-4, 2e-4, 4e-4};
+  auto deviations = m_config.deviations;
```

Comment:
Copy on purpose? If it's just to allow changing between `deviations` and `deviationsDisc`, maybe a pointer is better suited?